maryam/core/util/search/scholar.py

```diff
@@ -0,0 +1,71 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import lxml
+
+class main:
+
+	def __init__(self, q, limit=15):
+		""" scholar.google.com search engine
+
+			q	  : query for search
+			limit	  : maximum result count
+		"""
+		self.framework = main.framework
+		self.q = q
+		self.max = limit
+		self._articles = []
+		self._links_with_data = []
+
+	def run_crawl(self):
+		start = 0
+		self.q = self.framework.urlib(self.q).quote_plus
+		self.framework.verbose('Searching the scholar.google.com domain...')
+		for start in range(0, (self.max//10)+1):
+			url = f"https://scholar.google.com/scholar"
+			payload = {'as_vis': '1', 'q': self.q, 'start': start}
+			try:
+				req = self.framework.request(
+						url=url,
+						params=payload)
+			except:
+				self.framework.error('[SCHOLAR] ConnectionError', 'util/scholar', 'run_crawl')
+				self.framework.error('Google Scholar is missed!', 'util/scholar', 'run_crawl')
```

Comment:
util/search/scholar

Reply:
Corrected. But its still util/whatever is all modules?